### ISSUE TITLE
Enhancement: [consistent-type-assertions] support both assertionStyle options

### ISSUE THREAD

**Issue by mbursill** [typescript-eslint/typescript-eslint#9558]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/consistent-type-assertions/

### Description

The devs on my team prefer a mix of assertion styles. Sometimes "angle-bracket" is preferred and other times "as" is preferred. We still want to enforce objectLiteralTypeAssertions set to "never". A 3rd option for assertionStyle to allow both would be great.

### Fail

```TypeScript
const x = { foo: 1 } as T;
const x = <T>{ foo: 1 };

function bar() {
  return { foo: 1 } as T;
}
```

### Pass

```TypeScript
const x: T = { foo: 1 };
const y: U = x as T;
const z: U = <U>x;
```

**Comment by kirkwaiblinger** [comment]
related: #8921, #6605

**Comment by bradzacher** [comment]
> The devs on my team prefer a mix of assertion styles. Sometimes "angle-bracket" is preferred and other times "as" is preferred.

In the broader ecosystem - this is just not something that's common. By that I mean codebases that mix the two syntaxes do so because there's code that hasn't been migrated, not that it's a conscious choice.

The vast, vast, vast majority of TS code uses `as` everywhere, and a small percentage uses angle bracket.

And ultimately that's the direction you should be going - pick `as`. Angle bracket assertions are not supported in tsx, cts or mts extensions.

---

This fails the "proposal would be useful to the broader TypeScript community" check for the above reason. And as noted in the above issues - it was an intentional change that was made many years ago.

It's just not something we're looking to change. The rule has a fixer which makes it easy to adopt one or the other for greater consistency and less code churn.
